src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -195,7 +195,14 @@ internal IList<object> GetMembersByInferredType(PSTypeName typename, bool isStat
                 // Look in the type table first.
                 if (!isStatic)
                 {
-                    var consolidatedString = new ConsolidatedString(new[] { typename.Name });
+                    // The Ciminstance type adapter adds the full typename with and without a namespace to the list of type names.
+                    // So if we see one with a full typename we need to also get the types for the short version.
+                    // For example: "CimInstance#root/standardcimv2/MSFT_NetFirewallRule" and "CimInstance#MSFT_NetFirewallRule"
+                    Match match = Regex.Match(typename.Name, "(Microsoft\\.Management\\.Infrastructure\\.CimInstance#)(.+\\/)(.+$)");
```

Comment:
I'm unsure if it's worth it but potentially you could avoid regex by seeing if it starts with `Microsoft.Management.Infrastructure.CimInstance#` then splitting the value after with `/` to build the short type name entry.

There's also a source generator for regex to have it compile the pattern automatically for you that may be of interest https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0.

Reply:
It is better not use regex (if possible) for security reasons.

Reply:
Sure, I'll remove the regex usage. What's the security problem with regex though?

Reply:
> What's the security problem with regex though?

I see the Regex is in "parser". So we must check if input data are checked enough to exclude _infinite_ Regex, for example.